modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageLeaderElectionListener.java

```diff
@@ -62,87 +66,108 @@ public class MetaStorageRaftGroupEventsListener implements RaftGroupEventsListen
      *
      * <p>Multi-threaded access is guarded by {@code serializationFutureMux}.
      */
+    @Nullable
     private CompletableFuture<Void> serializationFuture = null;
 
     private final Object serializationFutureMux = new Object();
 
     private final ClusterTimeImpl clusterTime;
 
-    MetaStorageRaftGroupEventsListener(
+    private final LogicalTopologyEventListener logicalTopologyEventListener = new MetaStorageLogicalTopologyEventListener();
+
+    private final MetaStorageConfiguration metaStorageConfiguration;
+
+    /**
+     * Leader term if this node is a leader, {@code null} otherwise.
+     *
+     * <p>Multi-threaded access is guarded by {@code serializationFutureMux}.
+     */
+    @Nullable
+    private Long thisNodeTerm = null;
+
+    MetaStorageLeaderElectionListener(
             IgniteSpinBusyLock busyLock,
             ClusterService clusterService,
             LogicalTopologyService logicalTopologyService,
             CompletableFuture<MetaStorageServiceImpl> metaStorageSvcFut,
-            ClusterTimeImpl clusterTime
+            ClusterTimeImpl clusterTime,
+            MetaStorageConfiguration metaStorageConfiguration
     ) {
         this.busyLock = busyLock;
         this.nodeName = clusterService.nodeName();
         this.logicalTopologyService = logicalTopologyService;
         this.metaStorageSvcFut = metaStorageSvcFut;
         this.clusterTime = clusterTime;
+        this.metaStorageConfiguration = metaStorageConfiguration;
     }
 
     @Override
-    public void onLeaderElected(long term) {
+    public void onLeaderElected(ClusterNode node, long term) {
+        System.err.println("FUCK: " + node + " " + term);
```

Comment:
Huh?

Reply:
Sorry, one of my debugging lines slipped in =(